### DISCUSSION TITLE
Is Saleor right for my use case?

### DISCUSSION THREAD

**Discussion by kk2211** [saleor/saleor#10265] [General]
Hi ,saleor looks awesome with good documentation and a good community . Our team is looking to use it for our e-commerce marketplace solution but we are not sure wether it is right for our us. Any advice would be helpful

Use Case:

We do require a multi-vendor marketplace, but not in a traditional sense. As in , the customers of our marketplace wont know who is the seller/vendor, they would just see that Our company is the one who is selling the product . 

We would be building two apps on top of Saleor Core:

- A standard e-commerce store with flutter
    - This would be similar to a single-vendor marketplace from Customers perspective
- An application or dashboard made specifically for each vendor , wherein  a vendor can
    - See basic metrics like total sales, number of orders,etc
    - Vendor can upload/delete products and maintain inventory

After skimming the docs, I see that we have a [Metadata API](https://docs.saleor.io/docs/3.x/developer/metadata) which we can use to add Metadata to Different objects. 

What we were thinking is that we can add a “Vendor” metadata key to each product which we can then use to query and build our Dashboards for each different vendor. Similarly we can use webhooks to update each vendor dashboard whenever an order is placed for one of their products.

Is this approach viable? Would love to hear some feedback/opinions before we start building our marketplace

**Comment by gauravraj-v** [comment]
Let me correct you there a bit. It doesn't have a great community. The core members hardly answer anything or help out on this forum. 
Coming to your use-case, metadata is a JSOB column in postgres. So you can certainly store vendor information in metadata. Querying speed will depending on defining the indexes on these fields. You'll have to configure that separately in postgres or django migrations. 
Conceptually, I'm assuming products will be independent of the vendor and one product can be sold by multiple vendors and vice versa. To handle this, you'll have to store an array of vendors in metadata and query them. Which seems unoptimized for SQL. 
Suggestions - 
1. If the number of vendors are not high, then you can store them as product attributes. Since attribute mappings are stored in a separate table, the queries should be simple enough. 
2. Make an App on top of saleor which uses the same database as Saleor but manages the Vendor table having reference to product table. This way you'll have access to the entire database, and you'll be able to customize anything. Try not to interfere with the saleor's core table schemas.

**Reply by looped-monk** [reply]
@gauravraj-v I just bumped across this discussion. I'm also looking to integrating Saleor as a marketplace model where vendors also need to have a dedicated dashboard with the relevant metrics, with relevant functionalities &  notifications configured as well. A user can be a customer as well as a vendor and Saleor(my side) has the functionality to decide the cut they take from the transaction. 

-I'm taking your suggestion of setting up an app and storing vendor id as product attributes. However, there is a pre-condition you mentioned "If the number of vendors are not high". In a typical setup with a growing user base , this could be high. 

-Regarding the dedicated dashboard for each vendor, is there any plugin or customization that you have come across ?

**Comment by kk2211** [comment]
Hi, Thank you for taking the time to respond.

Well its a bummer to hear this feedback about the community. At a glance the discussions and Issues sections seemed miles ahead of other e-commerce solutions (especially python based)

Coming to my use-case

-  95% of the products will have a unique vendor. Only some-odd products maybe in future might have multiple vendor but that will definitely be a rarity.

- If I understand correctly, are you saying that performance wise it would be better to use Attributes instead of MetaData? Is there more information about Database schemas/Design that Saleor implements? Would love to understand more. MetaData seems from the outset a better option because of the json nature and extensibility

**Reply by gauravraj-v** [reply]
What exactly will you store in metadata? Vendor ID and name? What if you want to store vendor information and other details in a separate table and then storing vendor ID reference in metadata won't make sense. But if you just want to store a single string/key value pair in metadata, then performance should be decent even with metadata. For metadata, you might have to explicitly define the index on the key you have stored.

**Reply by gauravraj-v** [reply]
Feature wise, Saleor is really ahead of the other open source solutions based on the features/functionalities. They have a very rapid rate of development as well.  But somehow don't get many responses on the community pages. They don't have enterprise support as well for self hosted solutions. 
- I would suggest doing a latency and concurrency test on your read/write requirements.
- Also think of an end-to-end solution using Saleor core and Apps which you want to develop.
- If you need free text search with facetting and other thing advanced search features, you might have to integrate with other 3rd party providers.
